### PR TITLE
add descriptor logic

### PR DESCRIPTION
**Pull request by gre**
add helpers to iterate on xpubs and descriptors.

also add `ledger-live scanDescriptors`

### CODE REVIEW

File: src/families/bitcoin/descriptor.test.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import invariant from "invariant";
+import { getCryptoCurrencyById } from "../../currencies";
+import {
+  inferDescriptorFromDeviceInfo,
+  inferDescriptorFromAccount,
+} from "./descriptor";
+import { fromAccountRaw } from "../../account";
+import bitcoinDatasets from "./datasets/bitcoin";
+
+describe("inferDescriptorFromAccount", () => {
```
Comment (onyb) [inline src/families/bitcoin/descriptor.test.js:11]: Can you add some tests for:
* wrapped segwit (`49'/0'/x'`)
* testnet accounts (`x'/1'/y'`)

File: src/derivation.js [hunk 2]
```diff
@@ -335,6 +346,20 @@ export const runDerivationScheme = (
...
+export const runAccountDerivationScheme = (
+  scheme: string,
+  currency: { coinType: number },
+  opts: {
+    account?: number | string,
+  } = {}
+) =>
+  runDerivationScheme(scheme, currency, {
+    ...opts,
+    address: "_",
```
Comment (juan-cortes) [inline src/derivation.js:359]: What's the logic for passing an underscore here 🤔

**Review by juan-cortes — APPROVED** [review]

File: src/derivation.js (hunk 2 above)
Comment (gre) [reply src/derivation.js:359]: We only want the account part of the derivation path, so I used _ as a way to identify which part is not account based. It's a bit overkill but it works

File: src/families/bitcoin/descriptor.test.js (hunk 1 above)
Comment (gre) [reply src/families/bitcoin/descriptor.test.js:11]: duly noted. At the moment however, we don't have segwit account in datasets/bitcoin and i'll wait @Arnaud97234  to provide one for future testing this.
we also don't have testnet datasets

as we try to keep these dataset static and from same QA see we'll add more test a bit later. this was tested on CLI and should work. There is also some chance that more functions will be needed when we will do the LLD integration, so this PR should be good to go as first kickoff of the logic.

**Pull request merged**